frontends/web/src/routes/account/send/send.tsx

```diff
@@ -16,33 +16,93 @@
 
 import { Component, h } from 'preact';
 import { route } from 'preact-router';
-import { translate } from 'react-i18next';
-import { isBitcoinBased } from '../utils';
-import { apiGet, apiPost } from '../../../utils/request';
-import { apiWebsocket } from '../../../utils/websocket';
-import { debug } from '../../../utils/env';
+import reject from '../../../assets/icons/cancel.svg';
+import approve from '../../../assets/icons/checked.svg';
+import { alertUser } from '../../../components/alert/Alert';
+import { Balance, BalanceInterface } from '../../../components/balance/balance';
 import { Button, ButtonLink, Checkbox, Input } from '../../../components/forms';
-import { Guide } from '../../../components/guide/guide';
 import { Entry } from '../../../components/guide/entry';
-import { store as fiat } from '../../../components/rates/rates';
-import { alertUser } from '../../../components/alert/Alert';
+import { Guide } from '../../../components/guide/guide';
 import { Header } from '../../../components/layout';
+import { store as fiat } from '../../../components/rates/rates';
 import Status from '../../../components/status/status';
 import WaitDialog from '../../../components/wait-dialog/wait-dialog';
-import { Balance } from '../../../components/balance/balance';
+import { translate, TranslateProps } from '../../../decorators/translate';
+import { debug } from '../../../utils/env';
+import { apiGet, apiPost } from '../../../utils/request';
+import { apiWebsocket } from '../../../utils/websocket';
+import { isBitcoinBased } from '../utils';
 import FeeTargets from './feetargets';
-import UTXOs from './utxos';
-import approve from '../../../assets/icons/checked.svg';
-import reject from '../../../assets/icons/cancel.svg';
 import * as style from './send.css';
+import { SelectedUTXOProps, UTXOs } from './utxos';
+
+interface SendProps {
+    accounts: AccountProps[];
+    code?: string;
+    deviceIDs: string[];
+}
+
+interface AccountProps {
+    code: string;
+    coinCode: string;
+}
+
+interface ProposedAmountProps {
+    amount: string | null;
```

Comment:
I would quite strongly prefer to use `amount?: string` (i.e. with the implicit `undefined`). In case you agree and just tried to model how the component was implemented before using TypeScript in this commit, then it's ok for me to leave this for a future commit.

Reply:
And on a different note: Shouldn't the logic be that *if* an `Amount` is provided, then the amount (and unit) field has to be set? It seems to me that you are making too many things optional here.